src/Schema/OracleSchemaManager.php

```diff
@@ -320,19 +323,226 @@ private function getQuotedIdentifierName($identifier)
         return $identifier;
     }
 
+    /**
+     * {@inheritdoc}
+     */
+    public function listTables()
+    {
+        $currentDatabase = (string) $this->_conn->getDatabase();
+
+        $columnsData     = $this->groupObjectRecordsByTable(
+            $this->getDatabaseColumns($currentDatabase),
+            'TABLE_NAME'
+        );
+        $foreignKeysData = $this->groupObjectRecordsByTable(
+            $this->getDatabaseForeignKeys($currentDatabase),
+            'TABLE_NAME'
+        );
+        $indexesData     = $this->groupObjectRecordsByTable(
+            $this->getDatabaseIndexes($currentDatabase),
+            'TABLE_NAME'
+        );
+
+        $tables = [];
+
+        foreach (array_keys($columnsData) as $tableName) {
+            $columns = $this->_getPortableTableColumnList(
+                $tableName,
+                '',
+                $columnsData[$tableName]
+            );
+
+            $foreignKeys = [];
+            if (! empty($foreignKeysData[$tableName])) {
+                $foreignKeys = $this->_getPortableTableForeignKeysList(
+                    $foreignKeysData[$tableName]
+                );
+            }
+
+            $indexes = [];
+            if (! empty($indexesData[$tableName])) {
+                $indexes = $this->_getPortableTableIndexesList(
+                    $indexesData[$tableName],
+                    $tableName
+                );
+            }
+
+            $table = new Table($tableName, $columns, $indexes, [], $foreignKeys, []);
+            if (! empty($columnsData[$tableName][0]['TABLE_COMMENTS'])) {
+                $table->addOption('comment', $columnsData[$tableName][0]['TABLE_COMMENTS']);
+            }
+
+            $tables[] = $table;
+        }
+
+        return $tables;
+    }
+
+    /**
+     * Gets all column definitions in one database call.
+     *
+     * @return array<int, array<string, mixed>>
```

Comment:
Please use `@return list<...>` instead of `@return array<int, ...>` in these methods. It's known that the array indexes are 0, 1, 2, etc., not just random integers.

Reply:
done